### ISSUE TITLE
`<String as der::Encode>::to_vec` can result in ambiguous code

### ISSUE THREAD

**Issue by Nemo157** [RustCrypto/formats#724]
If you have `Encode` imported in a module for other uses it could be easy to mess up `.to_vec()` vs `.into_bytes()`.

```rust
use der::Encode;

fn main() {
    assert_eq!("foo".to_string().into_bytes(), "foo".to_string().to_vec().unwrap());
}
```

```
thread 'main' panicked at 'assertion failed: `(left == right)`
  left: `[102, 111, 111]`,
 right: `[12, 3, 102, 111, 111]`', src/main.rs:4:5
```

**Comment by tarcieri** [comment]
Yeah, the ambiguity of the name `to_vec` has come up before, although we don't have an issue which proposed to change it.

The `Decode` trait has a [`from_der` method](https://docs.rs/der/0.6.0/der/trait.Decode.html#method.from_der). Perhaps it could be changed to `to_der` or `to_der_vec`.

**Comment by baloo** [comment]
I think this can be closed.
This was fixed by #898
